fabric-crash-report-info-v1/src/main/java/net/fabricmc/fabric/impl/crash/report/info/ThreadPrinting.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.crash.report.info;
+
+import java.lang.management.LockInfo;
+import java.lang.management.MonitorInfo;
+import java.lang.management.ThreadInfo;
+
+public class ThreadPrinting {
+	public static String fullThreadInfoToString(ThreadInfo threadInfo) {
```

Comment:
Please add a comment saying this was copied from ThreadInfo.toString(), I do wonder if there is a better way that doesnt require copy/pasting almost 100 lines from java 🤔

Reply:
comment is added now with link. Don't see a way to reuse code due to the java code using a static final field for their constant